app/src/main/java/com/brewthings/app/ui/screens/scanning/ScanningScreenViewModel.kt

```diff
@@ -65,13 +65,33 @@ class ScanningScreenViewModel : ViewModel(), KoinComponent {
     }
 
     private fun observeDatabase() {
+        val pills = mutableListOf<String>()
         repo.fromDatabase()
             .onEach { raptPills ->
                 screenState = screenState.copy(savedPills = raptPills)
+                raptPills.onEach { pill -> pills.add(pill.macAddress) }
+                observeBrews(pills)
             }
             .launchIn(viewModelScope)
     }
 
+    private fun observeBrews(pills: MutableList<String>) {
+        val brews = mutableListOf<List<Brew>>()
+            screenState = screenState.copy(brews = emptyList())
+        viewModelScope.launch {
+            pills.onEach { pill ->
+                brews.add(
+                    repo.getBrews(pill)
+                )
+            }
+            val uniqueBrews = brews.flatten()
+                .distinctBy { it.og.timestamp to it.fgOrLast.timestamp }
```

Comment:
serve davvero questa distinctBy?

Reply:
A quanto pare sì, senza mi scoppia.

Reply:
molto strano, conviene indagare

Reply:
Confermo, se tolgo la distinct scoppia.

FATAL EXCEPTION: main
                                                                                                    Process: com.brewthings.app.debug, PID: 13666
                                                                                                    java.lang.IllegalArgumentException: Key "Brew_2024-05-27T06:59:17.410Z" was already used. If you are using LazyColumn/Row please make sure you provide a unique key for each item.
                                                                                                    	at androidx.compose.ui.layout.LayoutNodeSubcompositionsState.subcompose(SubcomposeLayout.kt:437)
                                                                                                    	at androidx.compose.ui.layout.LayoutNodeSubcompositionsState$Scope.subcompose(SubcomposeLayout.kt:872)
                                                                                                    	at androidx.compose.foundation.lazy.layout.LazyLayoutMeasureScopeImpl.measure-0kLqBqw(LazyLayoutMeasureScope.kt:125)
                                                                                                    	at androidx.compose.foundation.lazy.LazyListMeasuredItemProvider.getAndMeasure(LazyListMeasuredItemProvider.kt:48)